mediaire_toolbox/queue/redis_wq.py

```diff
@@ -116,6 +191,15 @@ def lease(self, lease_secs=5, block=True, timeout=None):
             # Note: if we crash at this line of the program, then GC will
             # see no lease
             # for this item a later return it to the main queue.
+            while True:
+                if self._limit_rate(limit, timeunit):
+                    break
+                sleeptime = (timeunit == 'sec')*1\
```

Comment:
isn't this what the other inner method does?

Reply:
do you mean `_get_timestamp` mentioned above? Not quite the same, the logic here is that the queue should wait for sleep for different times if the time unit is different(1 sec if the timeunit is seconds, 10 sec if minutes and 1 minute if hours) On second thoughts We could also just set them to 1 sec regardless of the timeunit